internal/grpc/services/gateway/storageprovider.go

```diff
@@ -572,7 +571,12 @@ func (s *svc) InitiateFileUpload(ctx context.Context, req *provider.InitiateFile
 
 			// TODO(labkode): calculate signature of the whole request? we only sign the URI now. Maybe worth https://tools.ietf.org/html/draft-cavage-http-signatures-11
 			target := u.String()
-			token, err := s.sign(ctx, target)
+			ttl := time.Duration(s.c.TransferExpires) * time.Second
+			expiresAt := time.Now().Add(ttl).Unix()
+			if storageRes.Protocols[p].Expiration != nil {
+				expiresAt = utils.TSToTime(storageRes.Protocols[p].Expiration).Unix()
```

Comment:
Mmh. Why would the protocol expiration overwrite our upload expiration? How are they related? And who wins if they have different values?

Reply:
The expiration can now be configured per storage and this is how the expiration is propagated to the gateway. The expiration setting in the gateway is merely a fallback for storage providers that do not set their own expiration.